XMonad/Actions/Workscreen.hs

```diff
@@ -90,7 +93,7 @@ viewWorkscreen wscrId = do (WorkscreenStorage c a) <- XS.get
                            let wscr = if wscrId == c
                                           then Workscreen wscrId $ shiftWs (workspaces $ a !! wscrId)
                                           else a !! wscrId
-                               (x,_:ys) = splitAt wscrId a
+                               (x, nonEmpty -> _ :| ys) = splitAt wscrId a
```

Comment:
Since `wscrId` is just some random Int, this is probably not safe either.  It's not any less unsafe than the rest of the function though (see the `if-then-else` above) so...

Reply:
The intended usage of this is to bind it to a key, so it's completely acceptable that a keybinding raises an exception which is caught and logged, and then everything continues as if nothing happened.

Reply:
That's fair enough